plugin/pkg/up/up_test.go

```diff
@@ -0,0 +1,71 @@
+package up
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+)
+
+func TestUp(t *testing.T) {
+	// We check a target 5 times, then check if that really happened. After which fails should have been reset to 0.
+	const max = 5
+	wg := sync.WaitGroup{}
+	fails := int32(3)
+	hits := int32(0)
+	web := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, r *http.Request) {
+			atomic.AddInt32(&hits, 1)
+		}))
+	defer web.Close()
+
+	pr := New()
+	defer pr.Stop()
+	i := 0
+	wg.Add(1)
+	upfunc := func(s string) bool {
+		http.Get(s)
```

Comment:
your test could be a lot simpler if it didn't actually do an HTTP Get. You wouldn't need the `web` server at all. It doesn't matter to this "up" machinery what the upfunc actually does.